src/Wikidata.php

```diff
@@ -54,12 +54,14 @@ public function getWikisourceLangs( string $lang ): array {
 			$data = $this->fetch( $query );
 			$out = [];
 			foreach ( $data as $datum ) {
-				preg_match( '|https://([a-z-_]*)\.?wikisource\.org|', $datum['website'], $matches );
-				$subdomain = $matches[1];
-				if ( empty( $subdomain ) ) {
-					$subdomain = 'mul';
+				// Hard-code Multilingual Wikisource, to avoid issues with incubator Wikisources
+				// being given the same domain name as P856 (official website). T342520.
+				if ( str_ends_with( $datum['item'], 'Q18198097' ) ) {
+					$out['mul'] = $datum['label'];
+					continue;
 				}
-				$out[$subdomain] = $datum['label'];
+				preg_match( '|https://([a-z-_]*)\.?wikisource\.org|', $datum['website'], $matches );
+				$out[$matches[1]] = $datum['label'];
```

Comment:
`$matches[1]` seems a bit fragile to me if Wikidata data are invalid for some reason.

Reply:
Good point. I did initially try it with *Wikimedia language code* (P424), but that doesn't get the correct value for some, like syl. Although maybe https://www.wikidata.org/wiki/Q120844812 should not be called a Wikisource language edition yet, while it's in incubator.

If someone puts an invalid value for Official website then you're right, this would break, but then whatever we pull from Wikidata could get broken at some point so I'm not sure it's too different.

Reply:
Sorry, I have made this disucssion slip. Indeed, that's a good point. I am fine with merging this PR.